tool/gzip.go

```diff
@@ -0,0 +1,39 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: MIT
+
+package tool
```

Comment:
Why are we putting the persistent queue code in the `tool` package?

Reply:
yeah , I will change it to util folder, do you think this is properly?